lib/net.js

```diff
@@ -524,16 +524,8 @@ Socket.prototype._read = function(n) {
 
 Socket.prototype.end = function(data, encoding) {
   stream.Duplex.prototype.end.call(this, data, encoding);
-  this.writable = false;
   DTRACE_NET_STREAM_END(this);
   LTTNG_NET_STREAM_END(this);
-
-  // just in case we're waiting for an EOF.
-  if (this.readable && !this._readableState.endEmitted)
-    this.read(0);
-  else
-    maybeDestroy(this);
-
   return this;
```

Comment:
wow, this differs from node streams, as `end`  does not return `this`. We might want to add it to streams too.

Reply:
I'll send a follow up PR to this one, which will be a semver-major.